core/trino-main/src/main/java/io/trino/operator/aggregation/AggregatorFactory.java

```diff
@@ -66,7 +78,8 @@ public Aggregator createAggregator()
         else {
             accumulator = accumulatorFactory.createIntermediateAccumulator(lambdaProviders);
         }
-        return new Aggregator(accumulator, step, intermediateType, finalType, inputChannels, maskChannel);
+        List<Integer> aggregatorInputChannels = intermediateStateChannel.isEmpty() ? rawInputChannels : ImmutableList.of(intermediateStateChannel.getAsInt());
```

Comment:
`aggregatorInputChannels` has different purpose depending on context. Make params explicit. Otherwise it's hard to figure what means what in what context.